include/inexor/vulkan-renderer/wrapper/resource_descriptor.hpp

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <vulkan/vulkan_core.h>
+
+#include <string>
+#include <vector>
+
+namespace inexor::vulkan_renderer::wrapper {
+
+/// @brief A class for descriptor management.
+/// Shader access of data is managed through descriptors.
+/// Descriptors are organized in descriptor sets.
+/// Descriptors sets are described through their descriptor set layout.
+/// Descriptor sets are allocated from descriptor pools.
+class ResourceDescriptor {
+private:
+    std::string m_name;
+    VkDevice m_device{VK_NULL_HANDLE};
+    VkDescriptorPool m_descriptor_pool{VK_NULL_HANDLE};
```

Comment:
Hmm, the descriptor pool probably shouldn't be a member here, since we should have one per render thread, rather than each descriptor.

Reply:
We could later offer an overloaded constructor which just copies the descriptor pool.